autopayroll-CordaService/workflows/src/main/kotlin/com/autopayroll/flows/PaymentFlow.kt

```diff
@@ -0,0 +1,65 @@
+package com.autopayroll.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import com.autopayroll.contracts.MoneyStateContract
+import com.autopayroll.states.MoneyState
+import com.autopayroll.states.PaymentRequestState
+import net.corda.core.contracts.requireThat
+import net.corda.core.flows.*
+import net.corda.core.utilities.ProgressTracker
+import net.corda.core.transactions.TransactionBuilder
+import net.corda.core.flows.FlowSession
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.flows.FinalityFlow
+import net.corda.core.flows.CollectSignaturesFlow
+import net.corda.core.identity.CordaX500Name
+
+// *********
+// * Flows *
+// *********
+@InitiatingFlow
+@StartableByService
+class PaymentFlowInitiator : FlowLogic<SignedTransaction>() {
+    override val progressTracker = ProgressTracker()
+
+    @Suspendable
+    override fun call(): SignedTransaction{
+        // Initiator flow logic goes here
+
+        val notary = serviceHub.networkMapCache.notaryIdentities[0]
+        val wBStateList =  serviceHub.vaultService.queryBy(PaymentRequestState::class.java).states
+        val vaultState = wBStateList.get(wBStateList.size -1).state.data
+        val output = MoneyState(vaultState.amount.toInt(),vaultState.toWhom)
+
+        val transactionBuilder = TransactionBuilder(notary)
+        val commandData = MoneyStateContract.Commands.Pay()
+        transactionBuilder.addCommand(commandData, ourIdentity.owningKey, vaultState.toWhom.owningKey)
```

Comment:
Should it be a good idea to update the PaymentRequestState to add a sort of acknowledgement to the Finance node that the payment has been processed.

Reply:
will update